Mathlib/Algebra/Star/Positive.lean

```diff
@@ -0,0 +1,187 @@
+
+This file defines the `AddSubmonoid` of nonegative elements `StarOrderedRing.positive R` in a
+star ordered ring `R` with carrier `{ x : R | 0 ≤ x }` (in fact, the definition only requires
+`OrderedAddCommMonoid R` to allow for other use cases, but it is primarily intended for
+`StarOrderedRing R`). Equivalently, this is the `AddSubmonoid` generated by the elements of the
+form `star s * s`. Even though this technically corresponds to the nonnegative elements, the
+"positive" terminology is standard throughout the literature (at least in C⋆-algebra theory and
+operator theory), so we choose to prefer that.
+
+The advantage of using this over other definitions of positivity is that it allows us to unify
+several different concepts under a single umbrella. For example, one might be tempted to consider
+the collection of bounded linear operators on an Hilbert space and define positivity as
+`∀ x, 0 ≤ ⟪T x, x⟫`, but this doesn't generalize nicely to C⋆-algebras. Alternatively, one could
+define positivity as those self-adjoint elements with nonnegative spectrum, but this requires the
+algebra to be unital. Our definition herein conveniently sidesteps these issues.
+
+We place everything in the `StarOrderedRing` namespace in order to avoid collisions or confusion
+with other parts of the library.
+
+## Main definitions
+
+* `StarOrderedRing.positive R`: The additive submonoid of a star ordered ring `R` generated by
+  elements of the form `star s * s` for `s : R`.
+* `StarOrderedRing.positive.selfAdjoint : positive R →+ selfAdjoint R`: the identity map as an
+  `AddMonoidHom`.
+* `StarOrderedRing.positive.instAddMonoidWithOne`: Makes `positive R` into an `AddMonoidWithOne`
+* `StarOrderedRing.positive.instCanonicallyOrderedAddMonoid`: Makes `positive R` into a
+  `CanonicallyOrderedAddMonoid` which avoids duplication of some lemmas (e.g., `zero_le`)
+
+-/
+
+universe u
+
+variable {R : Type u}
+
+namespace StarOrderedRing
+
+/-- The positive elements of a star ordered ring, as an additive submonoid.
+
+Note that the type class assumptions for `StarOrderedRing.positive` are actually significantly
+weaker (just `OrderedAddCommMonoid`) so that it can be used in as many contexts as possible, but
+the primary intended use case is when `R` is a `StarOrderedRing`.
+
+In addiiton, we note that `StarOrderedRing.positive` is just an alias of `AddSubmonoid.nonneg`, but
+we choose to keep these separate for naming purposes and also because we will add many instances to
+this object (viewed as a type).
+
+Regarding naming of lemmas: whenever a lemma involves a hypothesis of the form `0 ≤ x`, it will be
+referenced as `nonneg` in the declaration name, and we always prefer `0 ≤ x` over `x ∈ positive R`,
+even though these are definitionally equal. However, when the hypothesis is `x : positive R` (note
+the `:` versus `∈`), the declaration name will use `positive`, not `pos`, so as to avoid confusion
+with `0 < x`. -/
+alias AddSubmonoid.nonneg ← positive
```

Comment:
My understanding is that `alias` is equivalent to:
```lean
def newName same_hypothesesas_as_oldName := oldName
```
so in particular it has `default` transparency, as opposed to `reducible`, which is what I would get with `abbrev`. I think that's what we want. And to allow for Moritz to maybe use unbounded operators, I have left out the hypothesis `StarOrderedRing`.

Reply:
I don't see what we gain by not making it be reducible.

Reply:
It prevents us from accidentally acquiring instances from `AddSubmonoid.nonneg` on `StarOrderedRing.positive`. My bet is that it's a non-issue, but potentially annoyingly difficult to debug if it ever does arise. Moreover, I would prefer not to see `AddSubmonoid.nonneg` as the result of some `whnfR` just because this is at reducible transparency. I'm not sure why being reducible would benefit us in this situation. I'm open to being wrong here.